keyboards/darkproject/kd87b_ltd/matrix.c

```diff
@@ -0,0 +1,87 @@
+/*

```

Comment:
> This PR features custom matrix scanning code with adaptive scan rate and anti-ghosting.

Due to the maintenance cost involved, this code should be removed and the core implementation used instead.

Reply:
This code is simple and efficient, it uses basic C constructions and PAL to improve it's reliability. It also takes into account the individual features of the structure and layout of the keyboard. The main algorithm of key scanning is robust and this is probably the only way to register keypresses for this hardware configuration.

Reply:
But is it necessary, or just an optimization?

Reply:
Now adding custom matrix is mandatory because it resolves problem with ghosting and phantom keypresses on some boards.
I will support this code and keep it up to QMK standarts.
This applies to both KD83a LTD and KD87b LTD.

Reply:
From my perspective, still needs to use core and wouldn't be accepted till the code is removed.

Reply:
QMK already has some options to detect and handle ghosting (`MATRIX_HAS_GHOST`), and phantom keypresses can be solved with debouncing.   So, I'm not sure why a custom matrix would be necessary.  And most of the comments really indicate that this is an optimization, rather than a requirement.